provenance.jq

```diff
@@ -0,0 +1,73 @@
+# input: "build" object (with "buildId" top level key)
+# output: list of image tags
+def tags:
+	[
+		.source.arches[].tags[],
+		.source.arches[].archTags[],
+		.build.img
+	]
+;
+
+# input: "tags" object with image digest and platform arguments
+# output: json object for in-toto provenance subject field
+def subjects($platform; $digest):
+	($digest | split(":")) as $splitDigest |
+	{
+		"name": ("pkg:docker/" + . + "?platform=" + $platform),
```

Comment:
You needed `@uri` on this platform value, right?

```suggestion
		"name": "pkg:docker/\(.)?platform=\($platform | @uri)",
```

Do we need it on the repo/tag name as well?  Also we need to replace the tag separator with `@`, right? which is kind of hard to do 100% correctly without writing a full "reference parser" :sob:  we can maybe cheat here and assume we'll never have a by-digest reference or a non-standard-port registry and do something like `sub(":"; "@")` but it's a little gross because it's so inaccurate and has several obvious edge cases :disappointed:

Reply:
we actually have `@uri` here https://github.com/docker-library/meta-scripts/pull/76/files#diff-10e5b6b785f9538d42d6b75a65f23340111a8d9427518b516760afcbb7df75e2R42 where the variable is assigned

the repo/tag name doesn't use the URL encoding AFAICT, we have this working e2e now with signing/verification

Reply:
fixed the string insertion